File: app/src/main/java/com/lingoswap/domain/repository/AuthRepository.java
package com.lingoswap.domain.repository;

import com.lingoswap.data.model.ApiResponse;
import com.lingoswap.data.model.AuthResponse;
import com.lingoswap.data.model.ResetPasswordRequest;
import com.lingoswap.data.model.request.RegisterRequest;
import com.lingoswap.data.repository.RepositoryCallback;

public interface AuthRepository {
    void login(String email, String password, RepositoryCallback<AuthResponse> callback);
    void register(RegisterRequest request, RepositoryCallback<AuthResponse> callback);
    void logout(RepositoryCallback<Void> callback);
    void forgotPassword(String email, RepositoryCallback<ApiResponse> callback);
    void resetPassword(ResetPasswordRequest request, RepositoryCallback<ApiResponse> callback);
}

File: app/src/main/java/com/lingoswap/domain/repository/AuthRepository.kt
package com.lingoswap.domain.repository

import com.lingoswap.data.model.request.LoginRequest
import com.lingoswap.data.model.request.RegisterRequest
import com.lingoswap.data.model.response.AuthResponse
import com.lingoswap.utils.Resource
import kotlinx.coroutines.flow.Flow

interface AuthRepository {
    suspend fun login(request: LoginRequest): Flow<Resource<AuthResponse>>
    suspend fun register(request: RegisterRequest): Flow<Resource<AuthResponse>>
    suspend fun forgotPassword(email: String): Flow<Resource<Unit>>
}
